test/unit/logic/transaction_pool.js

```diff
@@ -110,27 +170,1073 @@ describe('transactionPool', () => {
+			});
+
+			describe('contains 1 transaction', () => {
+				describe('that is valid', () => {
+					const validTransaction = { id: 'validTx' };
+					const transactions = [validTransaction];
+
+					before(done => {
+						transactionPool.addUnconfirmedTransaction(validTransaction);
+						transactionPool.getUnconfirmedTransactionList = function() {
+							return transactions;
+						};
+
+						undoUnconfirmedList((err, ids) => {
+							lastError = err;
+							lastIds = ids;
+							done();
+						});
+					});
+
+					it('should not return an error', () => {
+						return expect(lastError).to.not.exist;
+					});
+
+					it('should not log an error', () => {
+						return expect(logger.error.called).to.be.false;
+					});
+
+					it('should return valid ids array', () => {
+						expect(lastIds).to.be.an('array');
+						return expect(lastIds).to.deep.equal(
+							_.map(transactions, tx => {
+								return tx.id;
+							})
+						);
+					});
+
+					describe('modules.transactions.undoUnconfirmed', () => {
+						it('should be called onece', () => {
+							return expect(dummyUndoUnconfirmed.calledOnce).to.be.true;
+						});
+
+						it('should be called with transaction as parameter', () => {
+							return expect(dummyUndoUnconfirmed.args[0][0]).to.deep.equal(
+								validTransaction
+							);
+						});
+					});
+
+					describe('lists', () => {
+						let index;
+
+						describe('unconfirmed', () => {
+							it('index should be undefined', () => {
+								index = transactionPool.unconfirmed.index[validTransaction.id];
+								return expect(index).to.be.an('undefined');
+							});
+						});
+
+						describe('queued', () => {
+							it('index should be set', () => {
+								index = transactionPool.queued.index[validTransaction.id];
+								return expect(index).to.be.a('number');
+							});
+
+							it('transaction at index should match', () => {
+								return expect(
+									transactionPool.queued.transactions[index]
+								).to.deep.equal(validTransaction);
+							});
+						});
+
+						describe('multisignature', () => {
+							it('index should be undefined', () => {
+								index =
+									transactionPool.multisignature.index[validTransaction.id];
+								return expect(index).to.be.an('undefined');
+							});
+						});
+					});
+
+					after(resetStates);
+				});
+
+				describe('that results with error on modules.transactions.undoUnconfirme', () => {
+					const badTransaction = { id: 'badTx' };
+					const transactions = [badTransaction];
+					const error = 'undo error';
+
+					before(done => {
+						dummyUndoUnconfirmed = sinon.spy((transaction, cb) => {
+							return cb(error);
+						});
+						TransactionPool.__set__(
+							'modules.transactions.undoUnconfirmed',
+							dummyUndoUnconfirmed
+						);
+
+						transactionPool.addUnconfirmedTransaction(badTransaction);
+						transactionPool.getUnconfirmedTransactionList = function() {
+							return transactions;
+						};
```

Comment:
No need to call these functions in this unit test, everything should be stub or directly assigned